src/ServiceProviderInterface.php

```diff
@@ -13,10 +13,11 @@ interface ServiceProviderInterface
      * - the key is the entry name
      * - the value is a callable that will return the entry, aka the **factory**
      *
-     * Factories have the following signature:
-     *        function(\Psr\Container\ContainerInterface $container)
+     * A factory is an instance of {@see FactoryDefinitionInterface}, or a `callable` with the following signature:
+     * 
+     *     function(\Psr\Container\ContainerInterface $container)
```

Comment:
The `FactoryDefinitionInterface` as well as `callable` is already specified in the return type doc, and I hope we can add the callable signature there too. I feel we should describe what a factory does, instead. At the same time, it's useless to mention "container" here, because it's about service definitions, factories and extensions heere. I couldn't include some of the doc into this comment, but I think we should consider re-writing it in this way.

```suggestion
     * Retrieves the factories of this service provider.
     *
     * A factory is a function that can create and configure services.
```

Reply:
I try to use the more neutral term "entry" rather than "service", as not all entries are services.

Existing documentation in this doc-block also uses the term "entry".

Reply:
AFAIK, PSR-11 docs refer to anything that can be retrieved from the container as "service". I might need to check again. But yes, IMO it's a good idea to standardize these terms.

Reply:
> AFAIK, PSR-11 docs refer to anything that can be retrieved from the container as "service"

The written spec inconsistently uses these terms: predominantly "service", frequently "entry", and rarely "value".

The inline documentation exclusively uses "entry" - the term "service" is only used in conjunction with "provider".

Probably the spec should have started by defining an official terminology. It's probably outside the scope of this PR to start correcting it all though - we would need to agree on the official terminology first, and so on.

For now, to avoid growing the scope of this PR, I would suggest the inline documentation remain consistent with the terminology in the existing inline documentation, which exclusively uses the term "entry".

(Consider opening an issue to discuss/define the terminology issue in the future.)